subsys/logging/log_core.c

```diff
@@ -62,11 +64,121 @@ static k_tid_t proc_tid;
 static u32_t dummy_timestamp(void);
 static timestamp_get_t timestamp_func = dummy_timestamp;
 
+
+bool log_is_strdup(const void *buf);
+
 static u32_t dummy_timestamp(void)
 {
 	return 0;
 }
 
+/**
+ * @brief Count number of string format specifiers (%s).
+ *
+ * Result is stored as the mask (argument n is n'th bit). Bit is set if %s was
+ * found.
+ *
+ * @param str String.
+ * @param nargs Number of arguments in the string.
+ *
+ * @return Mask with %s format specifiers found.
+ */
+static u32_t count_s(const char *str, u32_t nargs)
+{
+	char curr = *str++;
+	bool arm = (curr == '%');
+	u32_t arg = 0;
+	u32_t mask = 0;
+	__ASSERT_NO_MSG(nargs <= 8*sizeof(mask));
+
+	while (curr) {
+		curr = *str++;
+		if (arm) {
+			if (curr == 's') {
+				mask |= BIT(arg);
+			}
+
+			if (curr != '%') {
+				arg++;
+			}
+
+			arm = false;
+		} else if (curr == '%') {
+			arm = true;
+		}
+
+		if (arg == nargs) {
+			return mask;
+		}
+	}
+
+	return mask;
+}
+
+/**
+ * @brief Check if address is in read only section.
+ *
+ * @param addr Address.
+ *
+ * @return True if address identified within read only section.
+ */
+static bool is_rodata(const void *addr)
+{
+#if defined(CONFIG_ARM) || defined(CONFIG_ARC) || defined(CONIFG_RISCV32) || \
```

Comment:
IMHO there should be generic kernel function for checking type of accessed area...
Could you please fill an issue requesting such feature?

Reply:
done: #16343